tests/test_evaluable.py

```diff
@@ -228,9 +228,10 @@ def test_dot(self):
         desired=numeric.contract(self.n_op_argsfun, self.shapearg, axis=iax),
         actual=evaluable.dot(self.op_args, self.shapearg, axes=iax))
 
+  @parametrize.skip_if(lambda skip_bool, **kwargs: skip_bool, "evaluable.sin requires arg.dtype ≥ float")
```

Comment:
Should `evaluable.sin` act on bool array? Do we need to come up some other test for `equal`, `greater` and `less`?

Reply:
For `function` I think we should try to follow numpy where we can, even if there does not seem to be any reasonable application. The way to go here would probably be an explicit cast, and have `evaluable.Sin` accept only float arguments. That's not to say we have to do all this now, though. I don't believe we have the cast operation yet, for example, so unless that is already on the todo list we could also keep things like these as they are. The main thing is that we don't introduce regressions.

Reply:
This test will fail due to type mismatch for e.g. the array object has `float` from `E.sin(E.eq(a, b))` and value has `float16` from `np.sin(np.equal(a, b))` which are 'not equal' as in the wrapped `evalf`. Same for `great` and `less`.

Reply:
If the issue is with `float16` vs `float`, isn't that handled by `evaluable.asdtype`?

Reply:
`float16` comes from applying `evaluable.Sin` to bool and in turn `np.sin` to bool. All the types are actually set correctly.

Reply:
```
    E.Equal   E.Sin
(a, a) -> bool -> float
   |       |       X
(a, a) -> bool -> float16
   np.equal  np.sin
```

Reply:
Or `asdtype(retval.dtype) == array.dtype` which comes down to the same thing - either way, we care about kind. I understood option Q4 to mean numpy's `can_cast` same kind, which unhelpfully allows casting of different kinds. But whether we call it Q1 or Q4, I hope we're on the same page now.

Reply:
Ah, I think i just got what you mean. Sorry for being so slow on this.

Reply:
Really, it didn't help that numpy made it confusing to use the word "same".